cmd/bnsd/client/client.go

```diff
@@ -333,16 +331,16 @@ func (b *BnsClient) SubscribeHeaders(out chan<- *Header) (func(), error) {
 // the given channel. If there is no error,
 // returns a cancel function that can be called to cancel
 // the subscription
-func (b *BnsClient) Subscribe(query tmpubsub.Query, out chan<- interface{}) (func(), error) {
+func (b *BnsClient) Subscribe(query tmpubsub.Query) (<-chan ctypes.ResultEvent, func(), error) {
 	ctx := context.Background()
-	err := b.conn.Subscribe(ctx, b.subscriber, query, out)
+	out, err := b.conn.Subscribe(ctx, b.subscriber, query.String())
```

Comment:
currently we are using the default which is 1. So that's why I didn't change any signatures, because behaviour is exactly the same as before - we get 1 by default. 

Not sure if we need this, but I can add. I have seen the optional parameter, just didn't figure that it's a nice to have.

Reply:
Agree it is a "nice to have". I will make another issue on it, you are right, nothing to do with tendermint upgrade

Reply:
Added #608